File: AndroidTools/AndroidTools/Domain/UseCases/Logcat/ClearLogcatUseCase.swift
//
//  ClearLogcatUseCase.swift
//  AndroidTools
//
//  Created by Thomas Bernard on 22/05/2024.
//

import Foundation

class ClearLogcatUseCase {
    private let logcatRepository : LogcatRepository = LogcatRepositoryImpl()
    
    func execute(deviceId : String) {
        logcatRepository.clearLogcat(deviceId: deviceId)
    }
}

File: lib/features/logcat/domain/usecases/clear_logcat_usecase.dart
import 'package:android_tools/features/logcat/domain/repositories/logcat_repository.dart';

class ClearLogcatUsecase {
  final LogcatRepository logcatRepository;
  const ClearLogcatUsecase({required this.logcatRepository});

  Future<void> call(String deviceId) {
    return logcatRepository.clearLogcat(deviceId);
  }
}
